activerecord/lib/active_record/connection_handling.rb

```diff
@@ -45,16 +45,20 @@ module ConnectionHandling
     # The exceptions AdapterNotSpecified, AdapterNotFound and +ArgumentError+
     # may be returned on an error.
     def establish_connection(spec = nil)
+      raise RuntimeError, "Anonymous class is not allowed." unless name
+
       spec     ||= DEFAULT_ENV.call.to_sym
       resolver =   ConnectionAdapters::ConnectionSpecification::Resolver.new configurations
-      spec     =   resolver.spec(spec)
+      # TODO: uses name on establish_connection, for backwards compatibility
+      spec     =   resolver.spec(spec, self == Base ? "primary" : name)
```

Comment:
Should we use `===` here for the comparison and not `==`? Think they map to `kind_of?` and `is_a?`, respectively, and that we don't want subclasses to hit the primary spec.

Also is this the reason for the optional `id` param in `spec`? Don't like changing the public API for this conditional.